src/main/java/com/parasoft/demoapp/grpc/service/JsonServiceImpl.java

```diff
@@ -1,9 +1,30 @@
 package com.parasoft.demoapp.grpc.service;
 
 
+import com.parasoft.demoapp.exception.ParameterException;
+import com.parasoft.demoapp.service.ItemInventoryService;
+import io.grpc.Status;
+import io.grpc.stub.StreamObserver;
 import net.devh.boot.grpc.server.service.GrpcService;
+import org.springframework.beans.factory.annotation.Autowired;
 
 @GrpcService
 public class JsonServiceImpl extends JsonServiceImplBase {
+    @Autowired
+    private ItemInventoryService itemInventoryService;
 
+    @Override
+    public void getStockByItemId(Long itemId, StreamObserver<Integer> responseObserver) {
+        try {
+            Integer inStock = itemInventoryService.getInStockByItemId(itemId);
+            responseObserver.onNext(inStock);
+            responseObserver.onCompleted();
+        } catch (ParameterException e) {
```

Comment:
This is only entered if the request 'itemId' is set to null.

However, when send request with null in SOAtest directly, the exception "Caused by: io.grpc.StatusRuntimeException: INTERNAL: Half-closed without a request" will be reported by gRPC directly, like below screenshot. And the getStockByItemId operation will not be performed - we can see no logs in the server's console.
![image](https://user-images.githubusercontent.com/109124579/210048267-a3f414a8-5ea5-4ee0-882c-dd918562b560.png)


In order to enter here for verifying its exception, we can manually set itemId with null in our code before calling 'getInStockByItemId', and send request with an Long type parameter in SOAtest, then we will get the exception like below but no response in Traffic Viewer (I'm not sure if it's a need to display here or not, if need, it needs to be researched):
![image](https://user-images.githubusercontent.com/109124579/210048294-de715dea-dd83-4e63-9f31-221883e10431.png)


For invalid format, like send request with String type parameter (and here what we required is Long type), it will be researched in seperate task.

Reply:
Thanks for the information. I've updated the description of the research task on these.